### ISSUE TITLE
Allow configuration of truthy keys

### ISSUE THREAD

**Issue by mondeja** [adrienverge/yamllint#344]
Linting Github Actions configuration files I'm getting the warning `truthy value should be one of [false, true]  (truthy)` with [`on` keys](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#on). For discussion, is not this default behaviour too much intrusive?

The solution would be set `check-keys: false` in `truthy` configuration, but this removes the verification for all keys. Maybe an `allowed-keys` configuration can help in this case?

Ignore solution if someone needs it:

```yaml
rules:
  truthy:
    allowed-values:
      - 'true'
      - 'false'
    check-keys: true
    ignore: |
      .github/workflows
```

**Comment by adrienverge** [comment]
Hello,

> The solution would be set `check-keys: false` in `truthy` configuration, but this removes the verification for all keys.

What do you mean exactly with "but this removes the verification for all keys"? Can you give an example of what values you would put in this `allowed-keys` list?

**Comment by mondeja** [comment]
> What do you mean exactly with "but this removes the verification for all keys"?

If I've another key, for example `On` in the file would not be checked, not?

- Can you give an example of what values you would put in this `allowed-keys` list?

```yaml
allowed-keys:
  - on
```

So with this `On` would be treated as warning but not `on`.

**Comment by adrienverge** [comment]
> If I've another key, for example `On` in the file would not be checked, not?

Sure, but why would you want to detect and report `On` (and other possibly truthy values), if you're not interested by reporting `on`? Why would you want to detect and report `On`, but not `Hey` or `Álvaro`?

If I understand your use-case, you either want to:
- use the rule `truthy` with `{check-keys: false}` (keys can be whatever they want),
- or have a new rule (not `truthy`) that would check keys and forbid them if they match a list.

**Comment by mondeja** [comment]
Sorry, but I don't understand your point. What is the solution that you propose for Github Actions configuration files linting? This next?

```yaml
rules:
  truthy:
    allowed-values:
      - 'true'
      - 'false'
    check-keys: true
    ignore: |
      .github
```

Ignore the file?

**Comment by adrienverge** [comment]
If you want to use `on` as a key in your YAML file, you need to disable the `truthy` rule or use `check-keys: false`. From what you post, it seems like you don't need `check-keys: true`.

**Comment by mondeja** [comment]
Wouldn't it be helpful if you could define the keys for which the rule works?

**Comment by adrienverge** [comment]
I don't think so, but maybe you have a specific use-case or example?

Why would you want to forbid `On` to be a key, but not `on`?

**Comment by mondeja** [comment]
This, for example:

```yaml
name: whatever

on:
  pull_request:
  push:
    branches:
      - master

jobs:
  whatever:
    runs-on: ubuntu-latest
    steps:
    - uses: some-new-action-that/someone-implements-in-the-future@v2
      with:
        yes: false
```

I don't know about you, but I don't want to be depending on this lack knowing that in the future the linting can fail.

**Comment by mondeja** [comment]
And why not add `fired` to be a truthy key that must be checked, or `released`, or `deprecated`? If this rule is so arbitrary, at least that keys can be configured.

**Comment by adrienverge** [comment]
> And why not add `fired` to be a truthy key that must be checked, or `released`, or `deprecated`? If this rule is so arbitrary, at least that keys can be configured.

Sure. As I said [here](https://github.com/adrienverge/yamllint/issues/344#issuecomment-734309493), what you're talking about is *a new rule* (not `truthy`) *that would check keys, and report is their value is forbidden*.
